<launch>
  <arg name="model" default="waffle" doc="model type [waffle, waffle_pi]"/>
  <arg name="use_robot_name1" default="robot1"/>
  <arg name="use_camera_name1" default="multi_camera"/>
  <arg name="init_pose" default="-x 0.0 -y 0.0 -Y 0"/>
  <arg name="d1" default="0.50"/>
  <arg name="h1" default="0.11"/>
  <arg name="xd" default="0.25" />
  <arg name="yd" default="0.00" />
  <arg name="zd" default="0.00" />

  <!-- These are the arguments you can pass this launch file, for example paused:=true -->
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>

  <include file="$(find omtb_gazebo)/launch/1tb_env1.launch">
    <arg name="debug" value="$(arg debug)" />
    <arg name="gui" value="$(arg gui)" />
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="headless" value="$(arg headless)"/>
  </include>
  <group ns="$(arg use_camera_name1)">
    <param name="tf_prefix" value="$(arg use_camera_name1)"/>
    <param name="robot_description"
      command="$(find xacro)/xacro --inorder '$(find multi_camera_description)/multi_camera_description/camera.urdf.xacro' d1:=$(arg d1) h1:=$(arg h1) xd:=$(arg xd) yd:=$(arg yd) zd:=$(arg zd)"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
       <param name="publish_frequency" type="double" value="100.0" />
    </node>
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf_spawner" respawn="false" output="screen"
    args="-urdf -model camera -x 0.0 -y 0.0 -Y 0.0 -param robot_description"/>
  </group>



  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg use_robot_name1)_tf_broadcaster" args="0 0 0 0 0 0 world $(arg use_robot_name1)/odom 100"/>
  <node pkg="tf" type="static_transform_publisher" name="world_to_$(arg use_camera_name1)_tf_broadcaster" args="0 0 0 0 0 0 world $(arg use_camera_name1)/base_footprint 100"/>

</launch>
